app/views/forms/_screen.html.erb

```diff
@@ -37,8 +37,101 @@
 
     </div>
 
-    <%# questions on the form %>
-    <%= index_table(Questioning, @form.questionings, :within_form => true) if form_mode != :new %>
+    <div class="questions">
+      <input class="btn btn-primary" type="submit" value="Save"></input>
+
+      <h2>Questions</h2>
+
+      <div class="questions-header" id="questions-header">
+        <span><a href="choose-questions"><i class="fa fa-plus"></i> Add Question</a></span>
+        <span class="q-addGroup"><a><i class="fa fa-folder-open-o"></i> Add Group</a></span>
+      </div>
+
+      <div class="draggable-list-wrapper">
+        <ol class="questions-list ui-sortable">
+          <% @form.questionings.each do |qing| %>
+            <li class="question">
+
+              <%# Standard question? %>
+              <% if qing.standardized? == true %>
+                <div class="q-standardized"><i class="fa fa-certificate"></i></div>
+              <% end %>
+
+              <%# Question type icon %>
+              <div class="q-type">
+                <% t = qing.qtype_name %>
+
+                <% if t == 'text' %>
+                  <span>abc</span>
+                <% elsif t == 'long_text' %>
+                  <i class="fa fa-align-left"></i>
+                <% elsif t == 'integer' %>
+                  <span>123</span>
+                <% elsif t == 'decimal' %>
+                  <span>1.23</span>
+                <% elsif t == 'select_one' %>
+                  <i class="fa fa-dot-circle-o"></i>
+                <% elsif t == 'select_multiple' %>
+                  <i class="fa fa-check-square-o"></i>
+                <% elsif t == 'date' %>
+                  <i class="fa fa-calendar"></i>
+                <% elsif t == 'time' %>
+                  <i class="fa fa-clock-o"></i>
+                <% elsif t == 'datetime' %>
+                  <i class="fa fa-calendar"></i>
+                  <i class="fa fa-clock-o"></i>
+                <% elsif t == 'location' %>
+                  <i class="fa fa-map-marker"></i>
+                <% end %>
+              </div>
```

Comment:
What I suggest you do here is simply `<%= form_item_icon(qing.qtype_name) %>`.

Then in forms_helper.rb you can put something like this:

```
FORM_ITEM_ICON_CLASSES = {
  'long_text' => 'fa fa-align-left',
  'date' => 'fa fa-calendar',
  ...
}

def form_item_icon(type)
  cls = FORM_ITEM_ICON_CLASSES[type]
  # Use font awesome icon if defined, else use custom icon from assets dir.
  cls ? content_tag(:i, class: cls) : image_tag("form_items/#{type}.png")
end
```

Again I know there is a lot going on here! Hopefully this is instructive for you.